instrumentation/lettuce/lettuce-5.1/src/test/groovy/io/opentelemetry/auto/instrumentation/lettuce/v5_1/LettuceReactiveClientTest.groovy

```diff
@@ -0,0 +1,473 @@
+            "net.peer.ip" "127.0.0.1"
+            "net.peer.port" port
+            "db.url" "redis://127.0.0.1:$port"
+            "db.type" "redis"
+            "db.statement" "key<TESTSETKEY> value<TESTSETVAL>"
+          }
+          event(0) {
+            eventName "redis.encode.start"
+          }
+          event(1) {
+            eventName "redis.encode.end"
+          }
+        }
+      }
+    }
+  }
+
+  def "get command with lambda function"() {
+    setup:
+    def conds = new AsyncConditions()
+
+    when:
+    reactiveCommands.get("TESTKEY").subscribe { res -> conds.evaluate { assert res == "TESTVAL" } }
+
+    then:
+    conds.await()
+    assertTraces(1) {
+      trace(0, 1) {
+        span(0) {
+          operationName "GET"
+          spanKind CLIENT
+          errored false
+          tags {
+            "net.transport" "IP.TCP"
+            "net.peer.ip" "127.0.0.1"
+            "net.peer.port" port
+            "db.url" "redis://127.0.0.1:$port"
+            "db.type" "redis"
+            "db.statement" "key<TESTKEY>"
+          }
+          event(0) {
+            eventName "redis.encode.start"
+          }
+          event(1) {
+            eventName "redis.encode.end"
+          }
+        }
+      }
+    }
+  }
+
+  // to make sure instrumentation's chained completion stages won't interfere with user's, while still
+  // recording metrics
+  def "get non existent key command"() {
+    setup:
+    def conds = new AsyncConditions()
+    final defaultVal = "NOT THIS VALUE"
+
+    when:
+    reactiveCommands.get("NON_EXISTENT_KEY").defaultIfEmpty(defaultVal).subscribe {
+      res ->
+        conds.evaluate {
+          assert res == defaultVal
+        }
+    }
+
+    then:
+    conds.await()
+    assertTraces(1) {
+      trace(0, 1) {
+        span(0) {
+          operationName "GET"
+          spanKind CLIENT
+          errored false
+          tags {
+            "net.transport" "IP.TCP"
+            "net.peer.ip" "127.0.0.1"
+            "net.peer.port" port
+            "db.url" "redis://127.0.0.1:$port"
+            "db.type" "redis"
+            "db.statement" "key<NON_EXISTENT_KEY>"
+          }
+          event(0) {
+            eventName "redis.encode.start"
+          }
+          event(1) {
+            eventName "redis.encode.end"
+          }
+        }
+      }
+    }
+
+  }
+
+  def "command with no arguments"() {
+    setup:
+    def conds = new AsyncConditions()
+
+    when:
+    reactiveCommands.randomkey().subscribe {
+      res ->
+        conds.evaluate {
+          assert res == "TESTKEY"
+        }
+    }
+
+    then:
+    conds.await()
+    assertTraces(1) {
+      trace(0, 1) {
+        span(0) {
+          operationName "RANDOMKEY"
+          spanKind CLIENT
+          errored false
+          tags {
+            "net.transport" "IP.TCP"
+            "net.peer.ip" "127.0.0.1"
+            "net.peer.port" port
+            "db.url" "redis://127.0.0.1:$port"
+            "db.type" "redis"
```

Comment:
`db.instance` and `db.statement` are marked as required in https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md

Reply:
Thanks, made sure statement is always filled (realized statement doc has a template for redis that includes the span name) and filed an issue to consider making instance optional since it seems like it wouldn't always apply (redis cluster has no db index concept anyways) https://github.com/open-telemetry/opentelemetry-specification/issues/658